src/App.ts

```diff
@@ -1029,18 +1029,22 @@ function singleAuthorization(
   authorization: Partial<AuthorizeResult> & { botToken: Required<AuthorizeResult>['botToken'] },
 ): Authorize<boolean> {
   // TODO: warn when something needed isn't found
-  const identifiers: Promise<{ botUserId: string; botId: string }> =
-    authorization.botUserId !== undefined && authorization.botId !== undefined
-      ? Promise.resolve({ botUserId: authorization.botUserId, botId: authorization.botId })
-      : client.auth.test({ token: authorization.botToken }).then((result) => {
-          return {
-            botUserId: result.user_id as string,
-            botId: result.bot_id as string,
-          };
-        });
+  let identifiers: { botUserId: string; botId: string };
 
   return async ({ isEnterpriseInstall }) => {
-    return { isEnterpriseInstall, botToken: authorization.botToken, ...(await identifiers) };
+    if (identifiers === undefined) {
```

Comment:
It looks like the `identifiers` here is always undefined. Is `let identifiers: { botUserId: string; botId: string };` necessary?

Reply:
This is to make the identifiers lazy: the first call of the returned anonymous function eveluates the identifiers.

Add to this, by now the `client.auth.test()` is in an async function, so you can catch its errors.

Reply:
In the previous version of the code, the architecture allows to create the promise and never await it: the promise is awaited only if you make a call of the asyhc function returns by `singleAuthorization`.

Reply:
Ah, alright. that makes sense (somehow I was confused here...). Thanks for explaining this 👍

Reply:
Probably, here, the call to `App.processEvent`¹  happends after that `client.auth.test()` emits error.

¹ it calls `this.authorize` that is returned by `singleAuthorization`